src/lib/y2partitioner/yast_nfs_client.rb

```diff
@@ -0,0 +1,264 @@
+    # information from the current devicegraph
+    #
+    # If the client is not available, it tries to install it first (with
+    # user confirmation).
+    #
+    # @return [Yast::Term, nil] a term defining the UI, nil if the client is
+    #   not available and was not installed
+    def init_ui
+      return nil unless try_to_ensure_client
+
+      configure_client
+      set_client_list
+      call_client("CreateUI")
+    end
+
+    # Handles an event coming from the UI, consolidating in the current
+    # devicegraph the changes reported by yast2-nfs-client
+    #
+    # @param event [Hash]
+    def handle_input(event)
+      return unless client_exists?
+
+      # It should be already done at this point, but just in case...
+      configure_client
+
+      widget_id = Yast::Event.IsWidgetActivated(event)
+      return if widget_id.nil?
+
+      client_result = call_client("HandleEvent", "widget_id" => widget_id)
+      # FIXME: Note copied from old partitioner (y2-storage) - Take care that
+      # non-fstab settings of nfs-client (firewall, sysconfig, idmapd) get
+      # written on closing partitioner
+      process_result(widget_id, client_result)
+    end
+
+    # Name of the package containing the YaST client
+    #
+    # @return [String]
+    def package_name
+      PACKAGE.dup
+    end
+
+  protected
+
+    # Checks whether the client is available and, if not, tries to install it of
+    # possible
+    def try_to_ensure_client
+      client_exists? || install_client
+    end
+
+    # Checks whether the client is available
+    #
+    # @return [Boolean]
+    def client_exists?
+      Yast::WFM.ClientExists(YAST_CLIENT)
+    end
+
+    # Installs the package that provides the required client, if possible
+    #
+    # The installation requires confirmation from the user.
+    #
+    # @return [Boolean] true if installation succeeded, false if it was not
+    #   possible to install the package or the user rejected to do it.
+    def install_client
+      if in_inst_sys?
+        log.info "Is not possible to install nfs-client"
+        return false
+      end
+
+      pkgs = [package_name]
+      log.info "Trying to install #{pkgs}"
+      Yast::PackageCallbacks.RegisterEmptyProgressCallbacks
+      res = Yast::PackageSystem.CheckAndInstallPackages(pkgs)
+      Yast::PackageCallbacks.RestorePreviousProgressCallbacks
+      log.info "Installation result: #{res}"
+      res
+    end
+
+    # Configures the client if it's possible and it's not already configured
+    def configure_client
+      return if self.class.client_configured?
+      # This is not an installed system, so no configuration to read
+      return if in_inst_sys?
+
+      log.info "Reading NFS settings"
+      call_client("Read")
+      self.class.mark_client_configured
+    end
+
+    # Checks whether the partitioner is being executed in the inst-sys (i.e.
+    # during system installation)
+    #
+    # @return [Boolean]
+    def in_inst_sys?
```

Comment:
Why not to use `Yast::Stage.initial`? I know it looks better, but just for consistence with other YaST code...

Reply:
Well, exactly because of that - it looks better. :-)

Reply:
In that case, I would rather prefer having that method in some place where other YaST code could use it (consistency + good looking). But for the time being, let's keep this as it is.